service/frontend/handler.go

```diff
@@ -273,15 +300,69 @@ func (wh *WorkflowHandler) UpdateDomain(ctx context.Context,
 	info := getResponse.Info
 	config := getResponse.Config
 	replicationConfig := getResponse.ReplicationConfig
-	clusterMetadate := wh.GetClusterMetadata()
-	currentCluster := clusterMetadate.GetCurrentClusterName()
-	isCurrentClusterActive := currentCluster == replicationConfig.ActiveClusterName
+	configVersion := getResponse.ConfigVersion
 	failoverVersion := getResponse.FailoverVersion
+	currentCluster := clusterMetadata.GetCurrentClusterName()
 
 	// whether active cluster is changed
 	activeClusterChanged := false
 	// whether anything other than active cluster is changed
 	configurationChanged := false
+
+	validateReplicationConfig := func(existingDomain *persistence.GetDomainResponse,
+		updatedActiveClusterName *string, updatedClusters []*gen.ClusterReplicationConfiguration) error {
+
+		if len(updatedClusters) != 0 {
+			configurationChanged = true
+			clusters := []*persistence.ClusterReplicationConfig{}
+			// this is used to prove that target cluster names is a superset of existing cluster names
+			targetClustersNames := make(map[string]bool)
+			for _, cluster := range updatedClusters {
+				clusterName := cluster.GetClusterName()
+				if err := wh.validateClusterName(clusterName); err != nil {
+					return err
+				}
+				clusters = append(clusters, &persistence.ClusterReplicationConfig{ClusterName: clusterName})
+				targetClustersNames[clusterName] = true
+			}
+			// validate that updated clusters is a superset of existing clusters
+			for _, cluster := range replicationConfig.Clusters {
+				if _, ok := targetClustersNames[cluster.ClusterName]; !ok {
+					return errCannotRemoveClustersFromDomain
+				}
+			}
+			replicationConfig.Clusters = clusters
+			// for local domain, the clusters should be 1 and only 1, being the current cluster
+			if len(replicationConfig.Clusters) > 1 && !existingDomain.IsGlobalDomain {
+				return errCannotAddClusterToLocalDomain
+			}
+		}
+
+		if updatedActiveClusterName != nil {
+			activeClusterChanged = true
+			if *updatedActiveClusterName != currentCluster {
+				errMsg := fmt.Sprintf("Can only set active cluster to current cluster: %s", currentCluster)
+				err := &gen.BadRequestError{Message: errMsg}
+				return err
+			}
+			replicationConfig.ActiveClusterName = currentCluster
+		}
+
+		// validate active cluster is also specified in all clusters
+		activeClusterInClusters := false
+		for _, cluster := range replicationConfig.Clusters {
+			if cluster.ClusterName == replicationConfig.ActiveClusterName {
+				activeClusterInClusters = true
+				break
```

Comment:
one coding practice we follow is to always use label when breaking out of loops.

Reply:
ok sure